data_structures/linked_list/add_nodes.py

```diff
@@ -0,0 +1,98 @@
+"""Adding two integers of same length using Linked lists
+
+Example add 321 + 248 = 569
+
+(1)->(2)->(3) + (8)->(4)->(2) = (5)->(6)->(9) """
+
+from doctest import testmod
+
+# node definition
+class Node:
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.next = None
+        # self.head=None
+
+
+# linkedlist definition
+class LL:
+    def __init__(self):
+        self.head = None
+
+    # method to print the linked list
+    def __str__(self) -> str:
+        temp = self.head
+        while temp:
+            print(temp.data, end=" ")
```

Comment:
Line 23 promises that this function will return a str instead of printing the str.

Reply:
working on it, will change